### PR TITLE
feat(data-exploration): table view improvements

### LINKED ISSUE #13223: Regressions in new data table component
## Bug description

Noticed a few problems in the new feature-flagged events table in Persons (e.g. on [this marius@posthog.com page](https://app.posthog.com/person/pI40e5w84oZfhqdX4Sqf9XVmH5vVM3pyOMY1KRfe9XT#activeTab=events)):
1. Property value columns don't show anything
1. There's no "Time" column
1. There's a redundant "Person" column when on the person page
1. Similarly, there's no date separators
1. _Also_, the ClickHouse query made by the new UI takes very long (literally 20 s on the person page linked above for me, as opposed to 0.5 s with the old UI), because there no longer is a minimum timestamp filter to avoid going through too many CH parts in one query
1. The event buffer notice is missing ("Note that some events with a never-before-seen distinct ID are deliberately delayed by 60 seconds ..."
1. "More"/kebab button column is too wide, it should have `width: 0`

### `data-exploration-live-events` disabled
on

### `data-exploration-live-events` enabled
off

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

Resolves https://github.com/PostHog/posthog/issues/13223

Implement several improvements to the data tables:

- [x] Bug [reported](https://posthog.slack.com/archives/C045L1VEG87/p1672767066838199) in slack
- [x] Add pagination
- [x] Get date rows back
- [x] Add tests for `dataNodeLogic` and `dataTableLogic`

image

Outside of the scope
- [ ] Restore team-level saved columns
- [ ] Add visual regression tests for tables with different config

## How did you test this code?

Wrote a lot of tests.

### CODE REVIEW

**Review by thmsobrmlr — APPROVED** [review]
LGTM

File: frontend/src/queries/nodes/DataTable/dataTableLogic.ts [hunk 1]
```diff
@@ -1,32 +1,93 @@
...
+                                        !dayjs(result[orderKeyIndex]).isSame(lastResult[orderKeyIndex], 'day')
+                                    ) {
+                                        newResults.push({
+                                            [categoryRowKey]: dayjs(result[orderKeyIndex]).format('LL'),
+                                        })
+                                    }
+                                    newResults.push(result)
+                                    lastResult = result
+                                }
+                                return newResults
```
Comment (Twixes) [inline frontend/src/queries/nodes/DataTable/dataTableLogic.ts:81]: I may be being dense, but I don't quite see why we're doing this at a glance. Can you add a comment for the future traveler?

Comment (mariusandra) [reply frontend/src/queries/nodes/DataTable/dataTableLogic.ts:81]: To get those date rows that you blocked me on before :D... or what do you mean?

Comment (mariusandra) [reply frontend/src/queries/nodes/DataTable/dataTableLogic.ts:81]: I renamed it to `resultsWithLabelRows` and added a comment.

**Pull request merged**